### PR TITLE
bump go to 1.22, k8s deps to 0.30.0

### PR DESCRIPTION
**Pull request by everettraven**
**Description of the change:**
- Bumps go to 1.22
- Bumps k8s deps to 0.30.0
- Bumps controller-runtime to 0.18.2
- Updates `handler` package so that handlers satisfy the updated controller-runtime APIs that use generics

**Motivation for the change:**
- support go 1.22 and k8s 1.30

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+	github.com/operator-framework/api v0.24.0
 	github.com/prometheus/client_golang v1.19.0
 	github.com/prometheus/client_model v0.6.1
-	k8s.io/api v0.29.2
-	k8s.io/apimachinery v0.29.2
-	k8s.io/client-go v0.29.2
+	k8s.io/api v0.30.0
+	k8s.io/apimachinery v0.30.0
+	k8s.io/client-go v0.30.0
 	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
```
Comment (everettraven) [inline go.mod:15]: I _think_ I need to update this still

File: internal/annotation/filter.go [hunk 2]
```diff
@@ -138,60 +138,68 @@ func validateAnnotation(key string, truthy bool) error {
...
-	hdlr *handler.EnqueueRequestForObject
+	hdlr *handler.TypedEnqueueRequestForObject[T]
 }
 
 // Create implements predicate.Predicate.Create().
-func (f *filter) Create(evt event.CreateEvent) bool {
-	if evt.Object == nil {
+func (f *filter[T]) Create(evt event.TypedCreateEvent[T]) bool {
+	var obj client.Object = evt.Object
+	if obj == nil {
```
Comment (everettraven) [inline internal/annotation/filter.go:151]: This was needed to maintain the existing expectations of the API where a `nil` object in the event would result in returning a default filter value

File: go.mod (hunk 1 above)
Comment (everettraven) [reply go.mod]: https://github.com/operator-framework/operator-lib/pull/176/commits/662cd2c7843ec5bb902a4eaa77e924e725cc28d8

File: go.mod [hunk 3]
```diff
@@ -1,19 +1,19 @@
 module github.com/operator-framework/operator-lib
 
-go 1.21
+go 1.22.0
```
Comment (rashmigottipati) [inline go.mod:3]: Looks like this wasn't automatically set due to bingo. So perhaps `go 1.22`?

Comment (everettraven) [reply go.mod:3]: Ah good call, I did set this myself

Comment (everettraven) [reply go.mod]: Fixed in https://github.com/operator-framework/operator-lib/pull/176/commits/c6b3af11f9eb215131fe782978380c0d01ce33ed

Comment (everettraven) [reply go.mod:3]: Looks like `go mod tidy` resets it. reverted in https://github.com/operator-framework/operator-lib/pull/176/commits/b6409b8033ebec33df0c32c3951bac84132fa61f

**Comment by everettraven** [comment]
Squashed commits

**Review by rashmigottipati — APPROVED** [review]
/approve
/lgtm

**Pull request merged**